src/main/java/com/xebialabs/deployit/ci/DeployitNotifier.java

```diff
@@ -343,6 +357,16 @@ public AutoCompletionCandidates doAutoCompleteApplication(@QueryParameter final
         }
 
         public FormValidation doCheckApplication(@QueryParameter String credential, @QueryParameter final String value, @AncestorInPath AbstractProject<?, ?> project) {
+            if (project == null) { // no context
+                if (!project.hasPermission(Jenkins.ADMINISTER)) {
+                    return FormValidation.ok();
+                }
+            } else if (!project.hasPermission(Item.EXTENDED_READ)
+                    && !project.hasPermission(CredentialsProvider.USE_ITEM)) {
+                return FormValidation.ok();
+            }
+            project.checkPermission(AbstractProject.CONFIGURE);
```

Comment:
The line and those above make no sense. In short you want:

* No permissions: Can throw from `checkPermission`, nobody cares
* ExtendedRead: Return OK
* Configure: Do everything

Reply:
removed the code and implemented form validation by checking permissions